packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDomainMapAndImporter.cpp

```diff
@@ -98,16 +98,49 @@ namespace {
   //
   // UNIT TESTS
   //
-
+#define applyAndCheckResult( A, B, newMap) \
+  { \
+    typedef typename STS::magnitudeType MT; \
+    MT norm = ScalarTraits<MT>::zero (); \
+    \
+    /* Fill a random vector on the original map */ \
+    Vector<Scalar,LO,GO,Node> AVecX((A).getDomainMap()); \
+    AVecX.randomize(); \
+    \
+    /* Import this vector to the new domainmap */ \
+    Vector<Scalar,LO,GO,Node> BVecX((B).getDomainMap()); \
+    Tpetra::Import<LO,GO,Node> TempImport((A).getDomainMap(), (newMap));  \
+    BVecX.doImport(AVecX,TempImport,Tpetra::INSERT); \
+    \
+    /* Now do some multiplies */ \
+    Vector<Scalar,LO,GO,Node> AVecY((A).getRangeMap()); \
+    Vector<Scalar,LO,GO,Node> BVecY((B).getRangeMap()); \
+    (A).apply(AVecX,AVecY); \
+    (B).apply(BVecX,BVecY); \
+    \
+    BVecY.update (-STS::one (), AVecY, STS::one ()); \
+    norm = BVecY.norm2(); \
+    \
+    out << "Residual 2-norm: " << norm << endl \
+        << "Residual 1-norm: " << BVecY.norm1 () << endl \
+        << "Residual Inf-norm: " << BVecY.normInf () << endl; \
+    \
+    /* Macros don't like spaces, so we put the test outside the */ \
```

Comment:
are these comments still relevant now that the code has been rearranged?

Reply:
Yes, still relevant, but pedantic.  I removed them.